src/components/Editor/index.tsx

```diff
@@ -39,6 +39,87 @@ import {goToPositionAndCenter} from "./MonacoInstance/utils";
 import "./index.css";
 
 
+/**
+ * Gets the begin line number of the log event selected by mouse in editor.
+ *
+ * @param editor
+ * @param beginLineNumToLogEventNumRefCurrent
+ * @return the begin line number of the selected log event
+ */
+const getSelectedLogEventNum = (
+    editor: monaco.editor.IStandaloneCodeEditor,
+    beginLineNumToLogEventNumRefCurrent: BeginLineNumToLogEventNumMap
+): Nullable<number> => {
+    const selectedLineNum = editor.getPosition()?.lineNumber;
+    if ("undefined" === typeof selectedLineNum) {
+        return null;
+    }
+
+    const selectedLogEventNum = getMapValueWithNearestLessThanOrEqualKey(
+        beginLineNumToLogEventNumRefCurrent,
+        selectedLineNum
+    );
+
+    if (null === selectedLogEventNum) {
+        // Unable to find logEventLineNum from logEventNum because `beginLineNumToLogEventNum`
+        // is either uninitialized or holds the value from the last loaded page.
+        return null;
+    }
+
+    return selectedLogEventNum;
+};
+
+/**
+ * Handles copy log event action in the editor.
+ *
+ * @param editor
+ * @param beginLineNumToLogEventNumRefCurrent
+ */
+const handleCopyLogEventAction = (
+    editor: monaco.editor.IStandaloneCodeEditor,
+    beginLineNumToLogEventNumRefCurrent: BeginLineNumToLogEventNumMap
+) => {
+    const selectedLogEventNum = getSelectedLogEventNum(
+        editor,
+        beginLineNumToLogEventNumRefCurrent,
+    );
+
+    if (null === selectedLogEventNum) {
+        return;
+    }
+    const selectedLogEventLineNum = getMapKeyByValue(
+        beginLineNumToLogEventNumRefCurrent,
+        selectedLogEventNum
+    );
+
+    if (null === selectedLogEventLineNum) {
+        // unreachable
+        return;
+    }
+    const nextLogEventLineNum = getMapKeyByValue(
+        beginLineNumToLogEventNumRefCurrent,
+        selectedLogEventNum + 1
+    );
+
+    const model: Nullable<monaco.editor.ITextModel> = editor.getModel();
+    if (null === model) {
+        return;
+    }
+    const maxLineNum: number = model.getLineCount();
+    const startLineNumber: number = selectedLogEventLineNum;
+    const endLineNumberMaybeNegative: number = null === nextLogEventLineNum ?
+        maxLineNum - 1 :
+        nextLogEventLineNum - 1;
+    const endLineNumber: number = Math.max(startLineNumber, endLineNumberMaybeNegative);
+    const endMaxColumn: number = model.getLineMaxColumn(endLineNumber);
+
+    editor.setSelection(new monaco.Range(startLineNumber, 0, endLineNumber, endMaxColumn));
+
+    // Monaco editor uses `document.execCommand` instead of the Clipboard API to copy text.
+    // eslint-disable-next-line @typescript-eslint/no-deprecated
+    editor.getContainerDomNode().ownerDocument.execCommand("copy");
```

Comment:
how about
```suggestion
    const selectionRange = new monaco.Range(
        selectedLogEventLineNum,
        0,
        nextLogEventLineNum - 1,
        Infinity
    );

    editor.setSelection(selectionRange);
    editor.trigger(handleCopyLogEventAction.name, "editor.action.clipboardCopyAction", null);
```

in this approach:
1. we don't have to get the max column number of the last line in the range
2. as a result, we don't have to get the editor model
3. by triggering the copy action, we avoid using native (but deprecated) the `execCommand("copy")` API.